### PR TITLE
Verify medium checksum before starting VM or after copying image

### PR DESCRIPTION
**Pull request by rwx788**
See [poo#41414](https://progress.opensuse.org/issues/41414).

#### Verification runs
* [qemu check failed](http://f174.suse.de/tests/101#step/data_integrity/1)
* [qemu check matched](http://f174.suse.de/tests/110#)
* [hyperv](https://openqa.suse.de/tests/2497277)
* [xen](https://openqa.suse.de/tests/2497279#)

Testing for hyperv and xen.

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -121,6 +122,12 @@ sub init_main {
     set_defaults_for_username_and_password();
     setup_env();
     check_env();
+    # We need to check image only for qemu backend, for svirt we validate image
+    # after it is copied to the hypervisor host.
+    if (check_var('BACKEND', 'qemu') && data_integrity_is_applicable()) {
+        my $errors = verify_checksum();
+        set_var('CHECKSUM_FAILED', $errors) if $errors;
```
Comment (foursixnine) [inline lib/main_common.pm]: ~~Why not record info instead?~~ Would you mind to also diag $errors?

File: tests/installation/data_integrity.pm [hunk 2]
```diff
@@ -7,19 +7,19 @@
...
+use version_utils 'is_svirt_except_s390x';
 
 sub run {
-    my $dir_path;
-    $dir_path = '/var/lib/libvirt/images/' if check_var('BACKEND', 'svirt');
-    $dir_path = get_var('HYPERV_DISK', 'D:') . '\\cache\\' if check_var('VIRSH_VMM_FAMILY', 'hyperv');
-    verify_checksum($dir_path);
+    # If variable is set, we only inform about it
+    my $errors = get_var('CHECKSUM_FAILED');
+    record_info("Checksum", $errors, result => 'fail') if $errors;
```
Comment (foursixnine) [inline tests/installation/data_integrity.pm]: ~~Ah! That's why~~

File: lib/main_common.pm (hunk 1 above)
Comment (rwx788) [reply lib/main_common.pm]: We do that later to diag all error at once. This is a hack for qemu to display it on the job page.

Comment (foursixnine) [reply lib/main_common.pm]: @rwx788 right! Skipped that line :)

**Pull request merged**